packages/flutter/lib/src/painting/box_decoration.dart

```diff
@@ -68,6 +68,8 @@ import 'image_provider.dart';
 ///
 ///  * [DecoratedBox] and [Container], widgets that can be configured with
 ///    [BoxDecoration] objects.
+///  * [SliverDecoration],a widget that can be configured with [BoxDecoration]
+///    objects that is specialized for rendering with slivers.
```

Comment:
Seeing these together makes me wonder if we should call it `DecoratedSliver` and have it take a `SliverDecoration` to match convention. But I think all slivers start with the name sliver...

It feels a bit odd to pass a `BoxDecoration` to a `SliverDecoration` too, since boxes and slivers are separate layout protocols. BoxDecoration.shadow expects a list of BoxShadows, BoxDecoration.border expects a BoxBorder, BoxDecoration.shape expects a BoxShape...

What do you think if this `SliverDecoration` --> `SliverBoxDecorationAdapter`?

Reply:
Kind of reminds me of `SliverToBoxAdapter`...

Reply:
As we talked earlier today, I changed it to `DecoratedSliver` to be consistent with `DecoratedBox` and this should be ok since a `DecoratedSliver` takes in a `Decoration` which is not limited to `BoxDecoration`.

Reply:
Can you add the same mention on ShapeDecoration and Decoration